src/utils/hooks/useListApplications.tsx

```diff
@@ -0,0 +1,62 @@
+import { useEffect, useState } from 'react'
+import {
+  Application,
+  GetApplicationsQuery,
+  Template,
+  useGetApplicationsQuery,
+} from '../../utils/generated/graphql'
+import { ApplicationDetails } from '../types'
+
+const useListApplication = () => {
+  const [applications, setApplications] = useState<ApplicationDetails[] | undefined>()
+  const [loading, setLoading] = useState(true)
+  const [error, setError] = useState('')
+
+  const { data, loading: apolloLoading, error: apolloError } = useGetApplicationsQuery()
+
+  useEffect(() => {
+    if (apolloError) return
+    if (apolloLoading) return
+    // Check that only one tempalte matched
```

Comment:
```suggestion
   // Check that only one template matched
```
Who cares about spelling in comments, right? ;)

Reply:
: ) I do! Everytime someone reads it would have been annoying - and you end up adding as part of another PR very unrelated. Thanks!